packages/jest-runtime/src/ScriptTransformer.js

```diff
@@ -208,8 +208,13 @@ class ScriptTransformer {
 
       if (typeof processed === 'string') {
         transformed.code = processed;
-      } else {
+      } else if (processed && typeof processed.code === 'string') {
```

Comment:
Also `typeof processed === 'object'` will pass for arrays as well, not only objects.